### DISCUSSION TITLE
Beacon configuration (Neptune 4 Pro)

### DISCUSSION THREAD

**Discussion by JakeBooher** [OpenNeptune3D/OpenNept4une#159] [Ideas]
This is a work in progress, and still needs some adjustments, but for now it seems to be working okay.

# Base Setup

Using the following printhead: https://www.printables.com/model/828330-elegoo-neptune-4pro-fan-shroud-v5-with-beaconcarto

**Setup Steps**:

1. Mount the beacon, and run the USB cable. Can plug into a USB hub so you can have a webcam connected at the same time. I'm using this one: https://www.amazon.com/gp/product/B0BGXQJQCX
2. Install Beacon Klipper Module: `cd ~ && git clone https://github.com/beacon3d/beacon_klipper.git && ./beacon_klipper/install.sh`
3. Make configuration adjustments outlined below

Add to moonraker.conf to allow beacon module to auto update:
```
[update_manager beacon]
type: git_repo
channel: dev
path: ~/beacon_klipper
origin: https://github.com/beacon3d/beacon_klipper.git
env: ~/klippy-env/bin/python
requirements: requirements.txt
install_script: install.sh
is_system_service: False
managed_services: klipper
info_tags:
  desc=Beacon Surface Scanner
```

Add to user_settings.cfg:
```
# Beacon Configuration
[beacon]
serial: /dev/serial/by-id/usb-Beacon_Beacon_RevH_(serial number)-if00
x_offset: 0 # update with offset from nozzle on your machine
y_offset: 22 # update with offset from nozzle on your machine
mesh_main_direction: x
mesh_runs: 2

[gcode_macro Calibrate_Probe_Z_Offset]
gcode:
      G28
      BEACON_CALIBRATE

[screws_tilt_adjust]
screw1: 33, 182.05
screw2: 33, 12.05
screw3: 203, 12.05
screw4: 203, 182.05

[safe_z_home]
home_xy_position: 112.5,112.5

[stepper_z]
homing_retract_dist: 0

[bed_mesh]
zero_reference_position: 112.5,112.5
horizontal_move_z: 0
speed:500
mesh_min: 15, 20.0
mesh_max: 220.00, 210
```

If you have the Beacon RevH you can also define it for resonance testing / X axis input shaping in user_config.cfg:
```
[resonance_tester]
accel_chip: beacon
probe_points:
    112.5,112.5, 20
```

# Contact

In addition to the changes above, if you'd like to set your Z offset automatically using Contact you can also make the following changes:

In `printer.cfg` comment out the entire `[safe_z_home]` block, and remove the override for this block that you set in user_settings.cfg. This will be redefined within the beacon block.

To the `[beacon]` block you defined in user_settings.cfg, add the following:

```
contact_max_hotend_temperature: 180 # increase to probe at print temps
home_z_hop: 5
home_z_hop_speed: 30
home_xy_move_speed: 300
home_xy_position: 112.5,112.5
home_method: contact # use proximity for induction homing
home_method_when_homed: proximity # after initial calibration use induction
home_autocalibrate: unhomed # contact will calibrate beacon on first home
```

Finally, you'll need to make changes to your PRINT_START macro. I did this by redefining the entire thing in user_settings.cfg, not sure if there's a better way. My entire PRINT_START macro with the beacon settings are:

```
[gcode_macro PRINT_START]   
variable_small_print: False
gcode:
    Frame_Light_ON
    Part_Light_ON
    G92 E0 
    G90                       ; Use absolute coordinates
    BED_MESH_CLEAR    
  
    {% set BED_TEMP = params.BED_TEMP|default(60)|float %}
    {% set BED_HEAT_SOAK_MINUTES = params.BED_HEAT_SOAK_MINUTES|default(0)|float %}
    {% set BED_MESH = params.BED_MESH|default('adaptive')|string %} ; One of: adaptive (default), full, default (or any other value as the bed mesh profile name), none
    {% set EXTRUDER_TEMP = params.EXTRUDER_TEMP|default(200)|float %}

    {% set area_start = params.AREA_START|default('0,0')|string %}
    {% set area_end = params.AREA_END|default('0,0')|string %}
    
    {% set start_coords = area_start.split(',')|map('float')|list %}
    {% set end_coords = area_end.split(',')|map('float')|list %}    

    {% set center_min_x = 62.5 %}                                             ; define centre area bounds
    {% set center_min_y = 62.5 %}
    {% set center_max_x = 172.5 %}
    {% set center_max_y = 172.5 %}
    
    ; Check if print is within center area
    {% if start_coords[0] >= center_min_x and start_coords[1] >= center_min_y and end_coords[0] <= center_max_x and end_coords[1] <= center_max_y %}
      SET_GCODE_VARIABLE MACRO=PRINT_START VARIABLE=small_print VALUE=True
    {% else %}                                                               ; print extends beyond center, set temperatures for both beds
      SET_GCODE_VARIABLE MACRO=PRINT_START VARIABLE=small_print VALUE=False 
    {% endif %}

    SET_GCODE_OFFSET Z=0 ; BEACON
    G28 ; BEACON
    G0 Z2 ; BEACON
    SET_BED_TEMPERATURE TARGET={BED_TEMP}                           ; Heat Bed to target temp
    SET_HEATER_TEMPERATURE HEATER=extruder TARGET=150   ; BEACON - heat nozzle to probe temp
    TEMPERATURE_WAIT SENSOR=extruder MINIMUM=149 MAXIMUM=155  ; BEACON - wait for nozzle to get to target
    BED_TEMPERATURE_WAIT MINIMUM={BED_TEMP-2} MAXIMUM={BED_TEMP+4}  ; Waits until the bed reaches close to target
    {% if BED_HEAT_SOAK_MINUTES > 0 %}
    RESPOND MSG="Waiting {BED_HEAT_SOAK_MINUTES} minutes for the bed to settle."
    G4 P{BED_HEAT_SOAK_MINUTES * 60000}
    {% endif %}
    G28 Z METHOD=CONTACT CALIBRATE=1 ; BEACON - Set z offset, and calibrate

    CG28
    
    {% if BED_MESH == 'full' %}
    BED_MESH_CALIBRATE
    {% elif BED_MESH == 'adaptive' %}
    BED_MESH_CALIBRATE ADAPTIVE=1
    {% elif BED_MESH != 'none' %}
    BED_MESH_PROFILE LOAD={BED_MESH}
    {% endif %}
    G28 Z METHOD=CONTACT CALIBRATE=0 ; ADJUST OFFSET
    SET_GCODE_OFFSET Z_ADJUST=-0.07
    Smart_Park
    SET_FILAMENT_SENSOR SENSOR=filament_sensor ENABLE=1
    SET_HEATER_TEMPERATURE HEATER=extruder TARGET={EXTRUDER_TEMP}   ; Set and heat the final extruder temperature
    TEMPERATURE_WAIT SENSOR=extruder MINIMUM={EXTRUDER_TEMP-4} MAXIMUM={EXTRUDER_TEMP+10}  ; Wait for extruder to reach near target temperature
    LINE_PURGE                                                      ; KAMP Line Purge near print
    G92 E0                                                          ; Reset Extruder
    G1 Z2.0 F3000                                                   ; Move Z Axis up 
    M117 Printing           
```

If your PRINT_START macro is different because you're using a different printer, or you've made customizations, you can view the diff of the stock PRINT_START macro and my customized one here: https://www.diffchecker.com/bta9olJI/

**Comment by mlee12382** [comment]
You realize that the bottom v-roller screw runs directly through the exclusion zone for the probe, which is going to potentially cause any number of errors and accuracy issues right? It's great that you're trying and everything but this is something that you should be considering if you haven't already.
BTT Eddy in the stock probe location would be a much better option for the N4.

**Reply by TheMeanCanEHdian** [reply]
> You realize that the bottom v-roller screw runs directly through the exclusion zone for the probe, which is going to potentially cause any number of errors and accuracy issues right? It's great that you're trying and everything but this is something that you should be considering if you haven't already. BTT Eddy in the stock probe location would be a much better option for the N4.

This is incorrect, the beacon with these styles of mounts has more than enough clearance.
![20240527_185022](https://github.com/OpenNeptune3D/OpenNept4une/assets/16025103/f90067bf-7d8e-42b9-82f3-6b0f500dac75)

**Reply by Banabas** [reply]
Hello.
Beacon is now running.
The only thing missing is the Auto Z-Offset function in the start g-code which I have to add myself.
Has anyone already done this?
So far my start g-code is standard opennept4une

**Reply by virajmalia** [reply]
> You realize that the bottom v-roller screw runs directly through the exclusion zone for the probe, which is going to potentially cause any number of errors and accuracy issues right? It's great that you're trying and everything but this is something that you should be considering if you haven't already. BTT Eddy in the stock probe location would be a much better option for the N4.

Can you check #244 and see if it works for you?

**Comment by Banabas** [comment]
Here’s my start_macro.
If you find anything that’s not right, let me know?
I’m not really familiar with this kind of thing.


```
[gcode_macro PRINT_START]
variable_small_print: False
gcode:
    Frame_Light_ON
    Part_Light_ON
    G92 E0
    G90                       ; Use absolute coordinates
    G28     ; home axes
    G0 Z2   ; position beacon at 2mm for heat soak
    BED_MESH_CLEAR    
    SET_GCODE_OFFSET Z=0      ; Reset Z offset
    
    ; Set temperatures and mesh
    {% set BED_TEMP = params.BED_TEMP|default(60)|float %}
    {% set BED_MESH = params.BED_MESH|default('adaptive')|string %}
    {% set EXTRUDER_TEMP = params.EXTRUDER_TEMP|default(200)|float %}
    
    ; Set start and end coordinates for center area check
    {% set area_start = params.AREA_START|default('0,0')|string %}
    {% set area_end = params.AREA_END|default('0,0')|string %}
    
    {% set start_coords = area_start.split(',')|map('float')|list %}
    {% set end_coords = area_end.split(',')|map('float')|list %}

    {% set center_min_x = 62.5 %}
    {% set center_min_y = 62.5 %}
    {% set center_max_x = 172.5 %}
    {% set center_max_y = 172.5 %}
    
    ; Check if print is within center area
    {% if start_coords[0] >= center_min_x and start_coords[1] >= center_min_y and end_coords[0] <= center_max_x and end_coords[1] <= center_max_y %}
      SET_GCODE_VARIABLE MACRO=PRINT_START VARIABLE=small_print VALUE=True
    {% else %}
      SET_GCODE_VARIABLE MACRO=PRINT_START VARIABLE=small_print VALUE=False
    {% endif %}

    ; Preheat bed and nozzle
    SET_BED_TEMPERATURE TARGET={BED_TEMP}
    M140 S{BED_TEMP}
    M109 S160  ; preheat nozzle to probing temp
    M190 S{BED_TEMP}  ; wait for bed temp
    G4 P60000  ; let temps settle for 1 min

    G28 Z METHOD=CONTACT CALIBRATE=1  ; calibrate Z offset and beacon model hot
    #Z_TILT_ADJUST  ; balance towers or adjust bed tilt
    BED_MESH_CALIBRATE RUNS=2  ; calibrate bed mesh in scan mode

    G28 Z METHOD=CONTACT CALIBRATE=0  ; calibrate Z offset after mesh

    ; Bed mesh calibration
    {% if BED_MESH == 'full' %}
      BED_MESH_CALIBRATE
    {% elif BED_MESH == 'adaptive' %}
      BED_MESH_CALIBRATE ADAPTIVE=1
    {% elif BED_MESH != 'none' %}
      BED_MESH_PROFILE LOAD={BED_MESH}
    {% endif %}
    
    Smart_Park
    SET_FILAMENT_SENSOR SENSOR=filament_sensor ENABLE=1
    SET_HEATER_TEMPERATURE HEATER=extruder TARGET={EXTRUDER_TEMP}
    TEMPERATURE_WAIT SENSOR=extruder MINIMUM={EXTRUDER_TEMP-4} MAXIMUM={EXTRUDER_TEMP+10}

    ; Nozzle priming and printing start
    LINE_PURGE
    G1 Z2.0 F3000  ; move Z axis up
    SET_GCODE_OFFSET Z=0.028  ; add offset for hotend thermal expansion
    

    G92 E0  ; reset extruder
    M117 Printing    
```

**Comment by Pivot021** [comment]
Can anyone confirm if we have to use OpenNeptune in order for the Beacon probe to work?

**Comment by alldone** [comment]
hi, what kind of type have you buyed
![image](https://github.com/user-attachments/assets/888dea88-879d-4257-9e22-475e2add89ab)
low_profile or normal 

for neptune 4 pro ???

**Reply by Pivot021** [reply]
I have the 4 Plus printer, and bought the low profile Rev H. I printed a bracket to fit the Beacon, and it's mounted, but I can't figure out how to disable the old probe in the CFG files? I wasn't sure if this was doable without OpenNeptune?

**Reply by Banabas** [reply]
I took the normal version rev.H. The connection goes upwards. I removed the additional fan. It doesn't fit.

**Reply by Pivot021** [reply]
I don't have the same size fans as other 4 Plus printers for some reason, so none of the shrouds fit mine. I used this: https://www.printables.com/model/696279-elegoo-neptune-3-4-pro-plus-max-beacon-mount

It fit perfectly.

**Comment by DBSDBS** [comment]
Hello,

For dummies like me, can someone explain WHERE to install and HOW to do this, for the 2 point:
"Install Beacon Klipper Module: cd ~ && git clone https://github.com/beacon3d/beacon_klipper.git && ./beacon_klipper/install.sh"

Thank you in advance

**Reply by JakeBooher** [reply]
You have to connect to it either through serial, or SSH: https://github.com/OpenNeptune3D/OpenNept4une/wiki#--ssh-requires-ethernet-connection-to-a-routernetwork-switch

**Comment by Voviks9965** [comment]
Hello, how is it better than the standard one that is installed in Elegoo Neptune 4?
